src/util/compat.test.js

```diff
@@ -72,6 +73,43 @@ describe("util/compat", () => {
     });
   });
 
+  describe("compatibleParser", () => {
+    function parseCompatible(x, type, handlers) {
+      return compatibleParser(type, handlers).parseOrThrow(x);
+    }
+    it("throws on json missing a compatible header", () => {
+      expect(() => parseCompatible("foo", "unused", {})).toThrow(
+        "unable to unwrap compatible"
+      );
+    });
+    it("throws on json with an invalid compatible header", () => {
+      expect(() =>
+        parseCompatible([{type: "foo", version: 3}, {}], "unused", {})
+      ).toThrow("unable to unwrap compatible");
+    });
+    it("throws on json with the wrong type", () => {
+      expect(() =>
+        parseCompatible([{type: "foo", version: "1.0.0"}, {}], "bar", {})
```

Comment:
Please use `toCompat` here instead to avoid breaking the opaque-type
abstraction boundary, which makes the test brittle. Likewise below.

Reply:
Done.